backend/tests/test_encoding_worker_manager.py

```
@@ -588,22 +598,101 @@ def test_raises_last_start_error_when_no_capacity_errors(self, manager, mock_db,
+        # Retried the whole 2-candidate set across multiple passes before failing.
+        assert mock_compute.start.call_count > 2
+        assert sleep.called
+
+    def test_retries_whole_set_on_all_transient_then_succeeds(self, manager, mock_db, mock_compute):
+        """A brief all-zones 503 blip: first pass fails everywhere, retry succeeds."""
+        mock_instance = MagicMock()
+        mock_instance.status = "TERMINATED"
+        mock_compute.get.return_value = mock_instance
+
+        def _op_503():
+            op = MagicMock(); op.error_code = "503"; op.error_message = "SERVICE UNAVAILABLE"
```

Ruff (0.16.1)

[error] Multiple statements on one line (semicolon)

(E702)